repos/system_upgrade/common/actors/cloud/checkgrubenvtofile/libraries/checkgrubenvtofile.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import ConvertGrubenvTask, FirmwareFacts, HybridImage
+
+GRUBENV_BIOS_PATH = '/boot/grub2/grubenv'
+GRUBENV_EFI_PATH = '/boot/efi/EFI/redhat/grubenv'
+
+
+def process():
+    hybrid_image = next(api.consume(HybridImage), None)
+
+    if hybrid_image and is_bios() and is_grubenv_symlink_to_efi():
+        reporting.create_report([
+            reporting.Title(
+                'Azure hybrid (BIOS/EFI) image detected. "grubenv" symlink will be converted to a regular file'
+            ),
+            reporting.Summary(
+                'Leapp detected the system is running on Azure cloud, booted using BIOS and '
+                'the "/boot/grub2/grubenv" file is a symlink to "../efi/EFI/redhat/grubenv". In case of such a '
+                'hybrid image scenario GRUB is not able to locate "grubenv" as it is a symlink to different '
+                'partition and fails to boot. If the system needs to be run in EFI mode later, please re-create '
+                'the relative symlink again.'
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Groups([reporting.Groups.PUBLIC_CLOUD]),
+        ])
+
+        api.produce(ConvertGrubenvTask())
+
+
+def is_bios():
+    """
+    Check whether system is booted into BIOS
+    """
+
+    ff = next(api.consume(FirmwareFacts), None)
+    return ff and ff.firmware == 'bios'
+
+
+def is_grubenv_symlink_to_efi():
+    """
+    Check whether '/boot/grub2/grubenv' is a relative symlink to '/boot/efi/EFI/redhat/grubenv'.
+    """
+
+    is_symlink = os.path.islink(GRUBENV_BIOS_PATH)
+    realpaths_match = os.path.realpath(GRUBENV_BIOS_PATH) == os.path.realpath(GRUBENV_EFI_PATH)
```

Comment:
@dkubek this is actually something what should be done in scanner actor as we still have the policy that check actors should not interact in any way with the system.

Reply:
I am thinking about updating the HybridImage model.